compiler/rustc_middle/src/hir/mod.rs

```diff
@@ -121,6 +121,17 @@ impl<'tcx> AttributeMap<'tcx> {
     }
 }
 
+#[derive(Default, Encodable, Debug, HashStable)]
+pub struct ModuleItems {
+    // Use BTreeSets here so items are in the same order as in the
+    // list of all items in Crate
+    submodules: BTreeSet<LocalDefId>,
+    items: BTreeSet<ItemId>,
+    trait_items: BTreeSet<TraitItemId>,
+    impl_items: BTreeSet<ImplItemId>,
+    foreign_items: BTreeSet<ForeignItemId>,
+}
```

Comment:
This structure is certainly missing a comment describing what these sets mean.
It's far from being obvious that the sets include *all* items in the module, including items inside function bodies etc, not only direct children, and such sets only make sense as some partitioning for parallel processing.